src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  const cleanedSourceString = sourceString.replace(/\s*;\s*/g, ';').trim();
+  const strArray = cleanedSourceString.split(';');
+  const result = {};
+
+  strArray.forEach((str) => {
```

Comment:
The `forEach` loop is acceptable here according to the checklist. However, it's important to be aware that if the string is empty or contains only semicolons, the loop will still execute once with an empty string, which will incorrectly add an empty key to the result object. You should filter out any empty strings before the `forEach` loop.